sources/packages/backend/apps/api/src/services/application-restriction-bypass/application-restriction-bypass.service.ts

```diff
@@ -0,0 +1,54 @@
+import { Injectable } from "@nestjs/common";
+import {
+  ApplicationRestrictionBypass,
+  ApplicationStatus,
+  RecordDataModelService,
+} from "@sims/sims-db";
+import { DataSource, Not } from "typeorm";
+
+/**
+ * Service layer for application restriction bypasses.
+ */
+@Injectable()
+export class ApplicationRestrictionBypassService extends RecordDataModelService<ApplicationRestrictionBypass> {
+  constructor(dataSource: DataSource) {
+    super(dataSource.getRepository(ApplicationRestrictionBypass));
+  }
+
+  /**
+   * Returns all application restriction bypasses for a given application.
+   * @param applicationId id of the application to retrieve restriction bypasses.
+   * @returns application restriction bypasses.
+   */
+  async getApplicationRestrictionBypasses(
+    applicationId: number,
+  ): Promise<ApplicationRestrictionBypass[]> {
+    return this.repo.find({
+      select: {
+        id: true,
+        isActive: true,
+        application: { id: true },
```

Comment:
selecting application object and adding application to the relations is not required to be used in where.